src/app/clusters/resource-monitoring-server/CodegenIntegration.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster.h>
+#include <app/static-cluster-config/ActivatedCarbonFilterMonitoring.h>
+#include <app/static-cluster-config/HepaFilterMonitoring.h>
+#include <app/util/attribute-storage.h>
+#include <data-model-providers/codegen/ClusterIntegration.h>
+#include <data-model-providers/codegen/CodegenDataModelProvider.h>
+
+#include <array>
+#include <cstdint>
+#include <forward_list>
+#include <memory>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ResourceMonitoring;
+using namespace Protocols::InteractionModel;
+
+namespace {
+struct ListEntry
+{
+    template <typename... Args>
+    ListEntry(Args &&... args) : cluster(std::forward<Args>(args)...), registration(cluster)
+    {}
+
+    ResourceMonitoringCluster cluster;
+    ServerClusterRegistration registration;
+    ListEntry * next{ nullptr };
+};
+
+ListEntry * gActiveClusters{ nullptr };
```

Comment:
So the tradeoff here vs the "usual" statically allocated arrays is that you might use less memory, but might fail to allocate it at runtime, right?  Is that the right tradeoff?  Especially considering that the init functions that do the allocation here have no way to communicate failure....